.github/workflows/conformance.yml

```diff
@@ -0,0 +1,93 @@
+on:
+  push:
+    branches: master
+  pull_request:
+    branches: '*'
+
+name: IPFS Conformance
+
+jobs:
+  conformance-interface:
+    strategy:
+      fail-fast: false
+      matrix:
+        toolchain:
+        - rust: stable
+        #- rust: nightly
+
+        platform:
+        - target: x86_64-unknown-linux-gnu
+          host: ubuntu-latest
+          cross: false
+
+        - target: x86_64-apple-darwin
+          host: macos-latest
+          cross: false
+
+#         - target: x86_64-pc-windows-msvc
+#           host: windows-latest
+#           cross: false
+
+#         - target: armv7-linux-androideabi
+#           host: ubuntu-latest
+#           cross: true
+
+#         - target: aarch64-linux-android
+#           host: ubuntu-latest
+#           cross: true
+
+#         - target: armv7-apple-ios
+#           host: macos-latest
+#           cross: true
+
+#         - target: aarch64-apple-ios
+#           host: macos-latest
+#           cross: true
+
+       # - target: wasm32-unknown-unknown
+       #   host: ubuntu-latest
+       #   cross: true
+    env:
+      RUST_BACKTRACE: 1
+      CARGO_INCREMENTAL: 0
+      LLVM_CONFIG_PATH: /usr/local/opt/llvm/bin/llvm-config
+      NDK_HOME: /usr/local/lib/android/sdk/ndk-bundle
+      VCPKGRS_DYNAMIC: 1
+
+    runs-on: ${{ matrix.platform.host }}
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Checkout tools repo
+        uses: actions/checkout@v2
+        with:
+          repository: ipfs-rust/ipfs-rust-conformance
+          path: conformance
+
+      - name: Cache cargo folder
+        uses: actions/cache@v1
+        with:
+          path: ~/.cargo
+          key: conformance-testing
+
+      - name: Install rust toolchain
+        uses: hecrj/setup-rust-action@v1
+        with:
+          rust-version: stable
+
+      - name: Build
+        if: matrix.platform.cross == false
+        run: cargo build --workspace
+
+      - name: npm install
+        run: npm install
+        working-directory: ./conformance
```

Comment:
Results of the `npm install` should likely be cached right away.

We likely don't want this on platforms which cannot build `ipfs-http` or platforms which can build it (possibly android) but we'd require an interpreter to run the tests in (cross compiled).

As the same goes for the next three tasks, I was thinking that maybe these tasks should be limited to only certain platforms (rust tier 1 for example). Another job would be great, if it re-use the `ipfs-http` from the previous job but I don't know how to do that.